### PR TITLE
Study Participant Registration

### PR DESCRIPTION
**Pull request by JoshAddington**
#### Adds Study Participant registration
Creates both a User and a Participant profile for the registrant. The participant profile itself contains no PII and is related to the User profile, which contains the user's name and email address. 

`host/study/register/` => displays RegisterParticipantForm  which asks for first name, last name, email address, company affiliation, phone number, sector, years of experience, type of degree and field of degree. The user must then view and accept the consent terms. The registrant can only register once with a given email address, and cannot register without accepting consent terms.

Upon registration, the registrant will be taken to a page notifying them that they will be notified by email if they are selected for the study and showing their generated username, which is random fruit and 3 random numbers, and pin number, which is 6 random numbers.

### CODE REVIEW

**Comment by JoshAddington** [comment]
lowercase and no passionfruit. :+1:

File: project/myuser/views.py [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
+        ip_address = x_forwarded_for.split(',')[-1].strip()
+    else:
+        ip_address = request.META.get('REMOTE_ADDR')
+    return ip_address
+
+
+class RegisterParticipantFormView(FormView):
+    template_name = 'registration/register.html'
+    form_class = RegisterParticipantForm
+    success_url = '/thanks/'
```
Comment (shapiromatron) [inline project/myuser/views.py]: Would be better to use a `from django.core.urlresolvers.reverse_lazy`

File: project/myuser/views.py [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+
+    def post(self, request, *args, **kwargs):
+        ip_address = get_ip_address(self.request)
+        form_class = self.get_form_class()
+        form = self.get_form(form_class)
+        if form.is_valid():
+            form.ipaddress = ip_address
```
Comment (shapiromatron) [inline project/myuser/views.py]: I might reorganize this a little bit more. I'd pass the `ipaddress` in using the the `get_form_kwargs` method, and then pop it off and set to the self.instance in the form `__init__` - example here: https://github.com/shapiromatron/hawc/blob/master/project/epi/forms.py#L118
https://github.com/shapiromatron/hawc/blob/master/project/epi/views.py#L138

File: project/myuser/views.py [hunk 3]
```diff
@@ -1,3 +1,43 @@
...
+            form.save()
+            request.session['username'] = form.name
```
Comment (shapiromatron) [inline project/myuser/views.py]: With setting the session, I would do this in the `self.form_valid`;  then we don't need to override the whole `post` function

File: project/myuser/views.py [hunk 4]
```diff
@@ -1,3 +1,43 @@
...
+        else:
+            return self.form_invalid(form)
+
+
+class RegisterThankYouView(TemplateView):
+    template_name = "registration/thanks.html"
+
+    def get_context_data(self, **kwargs):
+        context = super(RegisterThankYouView, self).get_context_data(**kwargs)
+        context['participant'] = {'username': self.request.session['username'],
```
Comment (shapiromatron) [inline project/myuser/views.py]: Not sure this is needed; I think we can add `django.template.context_processors.request` to the site settings, then could we use `{{request.user.username}}` in the template? Example settings here:
https://github.com/shapiromatron/hawc/blob/master/project/hawc/settings/base.py#L51

File: project/templates/base.html [hunk 5]
```diff
@@ -14,11 +14,13 @@
     {% if debug %}
     <link rel="stylesheet/less" type="text/css" media="all" href="{{ STATIC_URL }}less/site.less">
     <link rel="stylesheet/less" type="text/css" media="all" href="{{ STATIC_URL }}less/cards.less">
+    <link rel="stylesheet/less" type="text/css" media="all" href="{{ STATIC_URL }}less/register.less">
     <script src="//cdnjs.cloudflare.com/ajax/libs/less.js/1.7.3/less.min.js"></script>
     {% else %}
     {% compress css %}
     <link rel="stylesheet" type="text/less" media="all" href="{{ STATIC_URL }}less/site.less">
     <link rel="stylesheet" type="text/less" media="all" href="{{ STATIC_URL }}less/cards.less">
+    <link rel="stylesheet" type="text/less" media="all" href="{{ STATIC_URL }}less/cards.less">
```
Comment (shapiromatron) [inline project/templates/base.html]: Should be `less/register.less` I think right?

File: project/myuser/views.py [hunk 6]
```diff
@@ -1,3 +1,43 @@
+from django.http import HttpResponseRedirect
 from django.shortcuts import render
+from django.views.generic import FormView, TemplateView
+
+from .forms import RegisterParticipantForm
 
 # Create your views here.
+def get_ip_address(request):
```
Comment (shapiromatron) [inline project/myuser/views.py]: Move this function to `utils.views`?

**Comment by shapiromatron** [comment]
@JoshAddington Minor edits everywhere; looks great overall.

**Comment by shapiromatron** [comment]
Looks great @JoshAddington, thanks!

**Pull request merged**